src/main/java/uk/gov/ons/census/casesvc/service/CaseReceiptService.java

```diff
@@ -17,51 +24,121 @@
   private final CaseService caseService;
   private final CaseRepository caseRepository;
 
+  private Map<Key, Rule> rules = new HashMap<>();
+
   public CaseReceiptService(CaseService caseService, CaseRepository caseRepository) {
     this.caseService = caseService;
     this.caseRepository = caseRepository;
+    setUpRules();
+  }
+
+  private void setUpRules() {
+    /*
+     This table is based on: https://collaborate2.ons.gov.uk/confluence/pages/viewpage.action?spaceKey=SDC&title=Receipting
+     The invalid rows are missing from here and are handled collectively by throwing a " does not map to any valid processing rule" RunTimeException
```

Comment:
Would it have been better to map to an 'invalid' type rule, which could have thrown the exception, to make the table more explicit?

Reply:
I was debating on this, can do if other people are keen

Reply:
Would we then have to list every possible invalid combination? I'm fine with leaving 'invalid' as the implicit default.

Reply:
Used a BiFunction in the end, rather than a extended Rule class